go/arrow/ipc/file_reader.go

```diff
@@ -430,13 +430,18 @@ func (src *ipcSource) fieldMetadata(i int) *flatbuf.FieldNode {
 	return &node
 }
 
+func (src *ipcSource) variadicCount(i int) int64 {
+	return src.meta.VariadicCounts(i)
+}
+
 type arrayLoaderContext struct {
-	src     ipcSource
-	ifield  int
-	ibuffer int
-	max     int
-	memo    *dictutils.Memo
-	version MetadataVersion
+	src       ipcSource
+	ifield    int
+	ibuffer   int
+	ivariadic int
```

Comment:
maybe these should be grouped together into an embedded struct? (`iStruct` :chuckle:)

Reply:
since they are only used in the `arrayLoaderContext` i'm not sure what the benefit would be to doing so?

Reply:
a nit, so probably none